data_node/utils.go

```diff
@@ -1,3 +1,36 @@
 package datanode
 
+import (
+	"crypto/rand"
+	"fmt"
+	"os"
+
+	"github.com/SayedAlesawy/Videra-Ingestion/orchestrator/utils/errors"
+)
+
 //Here goes any utils that are specific to data node
+
+// createFileDirectory creates a directory with given permission
+func createFileDirectory(dirPath string, perm os.FileMode) error {
+	err := os.MkdirAll(dirPath, perm)
+	return err
+}
+
+func createFile(filepath string) error {
+	file, err := os.Create(filepath)
+
+	if errors.IsError(err) {
+		return err
+	}
+
+	defer file.Close()
+
+	return nil
+}
+
+// generateRandomString generates random string with length n
+func generateRandomString(n int) string {
+	b := make([]byte, n)
+	rand.Read(b)
```

Comment:
Would that only be digits? I think we can do a hex string. You will need to import `encoding/hex`
`
io.ReadFull(rand.Reader, b)
return hex.EncodeToString(b)
`

Reply:
it doesn't matter if it's digits or hex, I just wanted to avoid the confusion of expecting string with length n and find it's length 2n (as byte contains 2 hex values), and hence it will take some extra steps to make odd length string

Reply:
Okay, we can revisit this later if it caused problems.